include/deal.II/particles/particle_handler.h

```diff
@@ -1320,13 +1320,25 @@ namespace Particles
   inline void
   ParticleHandler<dim, spacedim>::serialize(Archive &ar, const unsigned int)
   {
-    // Note that we do not serialize the particle data itself. Instead we
-    // use the serialization functionality of the triangulation class, because
-    // this guarantees that data is immediately shipped to new processes if
-    // the domain is distributed differently after resuming from a checkpoint.
-    ar //&particles
-      &global_number_of_particles &global_max_particles_per_cell
-        &next_free_particle_index;
+    auto *ptr =
+      dynamic_cast<parallel::DistributedTriangulationBase<dim, spacedim> &>(
+        *triangulation);
+
+    if (ptr != nullptr)
+      {
+        // Note that, in the parallel case, we do not serialize the particle
+        // data itself. Instead we use the serialization functionality of the
+        // triangulation class, because this guarantees that data is immediately
+        // shipped to new processes if the domain is distributed differently
+        // after resuming from a checkpoint.
+        ar &global_number_of_particles &global_max_particles_per_cell
+          &next_free_particle_index;
+      }
+    else
+      {
+        ar &particles &global_number_of_particles &global_max_particles_per_cell
+          &next_free_particle_index;
+      }
```

Comment:
Yes, this is exactly what I had in mind. Would you mind writing this as follows, to avoid some code duplication:
```suggestion
    if (dynamic_cast<parallel::DistributedTriangulationBase<dim, spacedim> &>(
        *triangulation == nullptr)
      {
        // Note that, in the parallel case, we do not serialize the particle
        // data itself. Instead we use the serialization functionality of the
        // triangulation class, because this guarantees that data is immediately
        // shipped to new processes if the domain is distributed differently
        // after resuming from a checkpoint.
        ar &particles;
      }

    ar &global_number_of_particles &global_max_particles_per_cell
          &next_free_particle_index;
      }
```